### DISCUSSION TITLE
LESSON 11: failed to resolve versions

### DISCUSSION THREAD

**Discussion by kinjun751** [Cyfrin/foundry-full-course-cu#2444] [General]
Guys Im getting this error when I compile the deploy script, what do I do

failed to resolve versions
Found incompatible Solidity versions:
script/DeployBasicNft.s.sol ^0.8.20 imports:
    lib/forge-std/src/Script.sol >=0.6.2, <0.9.0
    src/BasicNft.sol =0.8.18
    lib/forge-std/src/console.sol >=0.4.22, <0.9.0
    lib/forge-std/src/console2.sol >=0.4.22, <0.9.0
    lib/forge-std/src/safeconsole.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdChains.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdCheats.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdJson.sol >=0.6.0, <0.9.0
    lib/forge-std/src/StdMath.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStorage.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStyle.sol >=0.4.22, <0.9.0
    lib/forge-std/src/StdUtils.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Base.sol >=0.6.2, <0.9.0
    lib/openzeppelin-contracts/contracts/token/ERC721/ERC721.sol ^0.8.20
    lib/forge-std/src/console.sol >=0.4.22, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStorage.sol >=0.6.2, <0.9.0
    lib/forge-std/src/console2.sol >=0.4.22, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/interfaces/IMulticall3.sol >=0.6.2, <0.9.0
    lib/forge-std/src/mocks/MockERC20.sol >=0.6.2, <0.9.0
    lib/forge-std/src/mocks/MockERC721.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStorage.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/openzeppelin-contracts/contracts/token/ERC721/IERC721.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC721/IERC721Receiver.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC721/extensions/IERC721Metadata.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/Context.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/Strings.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/introspection/ERC165.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/interfaces/draft-IERC6093.sol ^0.8.20
    lib/forge-std/src/interfaces/IERC20.sol >=0.6.2
    lib/forge-std/src/interfaces/IERC721.sol >=0.6.2
    lib/openzeppelin-contracts/contracts/utils/introspection/IERC165.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC721/IERC721.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/math/Math.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/math/SignedMath.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/introspection/IERC165.sol ^0.8.20
    lib/forge-std/src/interfaces/IERC165.sol >=0.6.2
Found incompatible Solidity versions:
src/BasicNft.sol =0.8.18 imports:
    lib/openzeppelin-contracts/contracts/token/ERC721/ERC721.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC721/IERC721.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC721/IERC721Receiver.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC721/extensions/IERC721Metadata.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/Context.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/Strings.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/introspection/ERC165.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/interfaces/draft-IERC6093.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/introspection/IERC165.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC721/IERC721.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/math/Math.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/math/SignedMath.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/introspection/IERC165.sol ^0.8.20

**Comment by EngrPips** [comment]
script/DeployBasicNft.s.sol ^0.8.20 imports:  src/BasicNft.sol =0.8.18 , why is your `DeployBasicNft` and `BasicNft` file using different version of compiler?

**Reply by kinjun751** [reply]
thanks Ive corrected it

**Reply by EngrPips** [reply]
You are welcome.
